### PR TITLE
Add CodeInfo rpc query

### PR DESCRIPTION
**Pull request by pinosu**

### CODE REVIEW

File: proto/cosmwasm/wasm/v1/query.proto [hunk 1]
```diff
@@ -65,6 +65,12 @@ service Query {
     option (google.api.http).get = "/cosmwasm/wasm/v1/code";
   }
 
+  // CodeInfo gets the metadata for a single wasm code
+  rpc CodeInfo(QueryCodeInfoRequest) returns (CodeInfoResponse) {
```
Comment (webmaster128) [inline proto/cosmwasm/wasm/v1/query.proto:69]: Should we have a dedicated type here callen `QueryCodeInfoResponse` for consistency? Then we could also use the opportunity to clean it up a little bit

File: x/wasm/client/cli/query.go [hunk 2]
```diff
@@ -252,20 +252,17 @@ func GetCmdQueryCodeInfo() *cobra.Command {
 			}
 
 			queryClient := types.NewQueryClient(clientCtx)
-			res, err := queryClient.Code(
+			res, err := queryClient.CodeInfo(
```
Comment (webmaster128) [inline x/wasm/client/cli/query.go:255]: Oh nice, this makes this query much more efficient. No more downloading and discarding the Wasm blob

**Review by webmaster128 — COMMENTED** [review]
Nice

cc @srdtrk

File: proto/cosmwasm/wasm/v1/query.proto [hunk 3]
```diff
@@ -222,6 +222,13 @@ message QueryCodeInfoRequest {
   uint64 code_id = 1; // grpc-gateway_out does not support Go style CodeID
 }
 
+// QueryCodeInfoResponse is the response type for the Query/Code RPC method
+message QueryCodeInfoResponse {
+  option (gogoproto.equal) = true;
+  CodeInfoResponse code_info = 1
+      [ (gogoproto.embed) = true, (gogoproto.jsontag) = "" ];
```
Comment (webmaster128) [inline proto/cosmwasm/wasm/v1/query.proto:229]: Can we inline all the fields and rename `DataHash` -> `Checksum`?

File: proto/cosmwasm/wasm/v1/query.proto [hunk 4]
```diff
@@ -208,7 +214,24 @@ message QuerySmartContractStateResponse {
...
-  uint64 code_id = 1; // grpc-gateway_out does not support Go style CodID
+  uint64 code_id = 1; // grpc-gateway_out does not support Go style CodeID
+}
+
+// QueryCodeInfoRequest is the request type for the Query/Code RPC method
+message QueryCodeInfoRequest {
+  uint64 code_id = 1; // grpc-gateway_out does not support Go style CodeID
+}
+
+// QueryCodeInfoResponse is the response type for the Query/Code RPC method
```
Comment (chipshort) [inline proto/cosmwasm/wasm/v1/query.proto:225]: I guess the RPC method in the doc comment should be adjusted

**Review by chipshort — COMMENTED** [review]
Nice!

**Review by chipshort — APPROVED** [review]
LGTM

**Pull request merged**